source/extensions/filters/udp/udp_proxy/udp_proxy_filter.cc

```diff
@@ -309,6 +309,14 @@ UdpProxyFilter::ActiveSession::~ActiveSession() {
       .connections()
       .dec();
 
+  // Clearing filters list to allow destructors of session filters to run
+  // and write to filter state before writing session access log.
+  if (Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.delete_session_filters_before_write_session_access_log")) {
+    write_filters_.clear();
```

Comment:
This is generally a very brittle solution, that can lead to use after free errors. UDP filters need `onDestroy` callback similar to HTTP filters and not rely on doing significant work in destructors. 

It maybe ok to leave a TODO here, even though this is a ticking time bomb.

Reply:
Thanks for the suggestion, I made a draft PR: [[UDP Proxy] Add onSessionComplete for UDP session filters #31094](https://github.com/envoyproxy/envoy/pull/31094)
Please, let me know if it's in the right direction.

Reply:
@yanavlasov WDYT?